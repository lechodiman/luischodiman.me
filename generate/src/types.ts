export interface Answers {
  title: string;
  description: string;
  draft: boolean;
  category: string;
  tags: string;
}
