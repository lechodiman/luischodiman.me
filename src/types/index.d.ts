import React from 'react';
import { FluidObject } from 'gatsby-image';

type RenderCallback = {
  render: (data: any) => React.ReactNode;
};

type Entry = {
  getIn: (data: string[]) => string;
};

type WidgetFor = (data: string) => string;

type PageContext = {
  tag: string;
  category: string;
  currentPage: number;
  prevPagePath: string;
  nextPagePath: string;
  hasPrevPage: boolean;
  hasNextPage: boolean;
};

type Node = {
  fields: {
    slug: string;
    categorySlug?: string;
    tagSlugs?: string[];
    editLink?: string;
  };
  frontmatter: {
    date: string;
    description?: string;
    category?: string;
    tags?: string[];
    title: string;
    socialImage: {
      sharp: {
        original: {
          src: string;
        };
        fluid: FluidObject;
      };
    };
    socialImageCredit: string;
  };
  html: string;
  id?: string;
};

type Edge = {
  node: Node;
};

type Edges = Edge[];

type AllMarkdownRemark = {
  allMarkdownRemark: {
    group: {
      fieldValue: string;
      totalCount: number;
    }[];
    edges: Edges;
  };
};

type MarkdownRemark = Node;
