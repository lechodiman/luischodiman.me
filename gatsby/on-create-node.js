'use strict';

const path = require('path');
const _ = require('lodash');
const { createFilePath } = require('gatsby-source-filesystem');

const onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions;

  if (node.internal.type === 'MarkdownRemark') {
    if (typeof node.frontmatter.slug !== 'undefined') {
      const dirname = getNode(node.parent).relativeDirectory;
      createNodeField({
        node,
        name: 'slug',
        value: `/${dirname}/${node.frontmatter.slug}`,
      });

      const editLink = path
        .normalize(node.fileAbsolutePath)
        .replace(path.join(__dirname, '..'), '')
        .replace(/\\/gi, '/');

      createNodeField({
        node,
        name: 'editLink',
        value: `https://github.com/lechodiman/luischodiman.me/edit/master${editLink}`,
      });
    } else {
      const value = createFilePath({ node, getNode });
      createNodeField({
        node,
        name: 'slug',
        value,
      });
    }

    if (node.frontmatter.tags) {
      const tagSlugs = node.frontmatter.tags.map(
        tag => `/tag/${_.kebabCase(tag)}/`
      );
      createNodeField({ node, name: 'tagSlugs', value: tagSlugs });
    }

    if (node.frontmatter.category) {
      const categorySlug = `/category/${_.kebabCase(
        node.frontmatter.category
      )}/`;
      createNodeField({ node, name: 'categorySlug', value: categorySlug });
    }
  }
};

module.exports = onCreateNode;
