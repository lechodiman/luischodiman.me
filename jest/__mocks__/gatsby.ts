import React from 'react';
import gatsby from 'gatsby';

export default {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(
      ({
        activeClassName,
        activeStyle,
        getProps,
        innerRef,
        ref,
        replace,
        to,
        ...rest
      }) =>
        React.createElement('a', {
          ...rest,
          href: to
        })
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
