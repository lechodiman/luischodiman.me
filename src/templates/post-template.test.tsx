import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import PostTemplate from './post-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import markdownRemark from '../../jest/__fixtures__/markdown-remark';
import { RenderCallback } from '../types';

const mockedUseStaticQuery = useStaticQuery as jest.Mock;
const mockedStaticQuery = StaticQuery as jest.Mock;

describe('PostTemplate', () => {
  const props = {
    data: {
      ...markdownRemark,
    },
  };

  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }: RenderCallback) =>
      render(siteMetadata)
    );
    mockedUseStaticQuery.mockReturnValue(siteMetadata);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<PostTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
