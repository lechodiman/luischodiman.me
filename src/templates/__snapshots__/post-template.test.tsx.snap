// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="post"
  >
    <a
      className="post__home-button"
      href="/"
    >
      Art√≠culos
    </a>
    <div
      className="post__content"
    >
      <div
        className="content"
      >
        <h1
          className="content__title"
        >
          test
        </h1>
        <div
          className="content__body"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>test</p>",
            }
          }
        />
      </div>
    </div>
    <div
      className="post__footer"
    >
      <div
        className="meta"
      >
        <p
          className="meta__date"
        >
          Publicado el 
          1 Sep 2016
        </p>
        <p
          className="meta__editLink"
        >
          <a
            rel="noopener noreferrer"
            target="_blank"
          >
            Editar post en Github
          </a>
        </p>
      </div>
      <div
        className="tags"
      >
        <ul
          className="tags__list"
        >
          <li
            className="tags__list-item"
          >
            <a
              className="tags__list-item-link"
              href="/test_0"
            >
              test_0
            </a>
          </li>
          <li
            className="tags__list-item"
          >
            <a
              className="tags__list-item-link"
              href="/test_1"
            >
              test_1
            </a>
          </li>
        </ul>
      </div>
      <div
        className="author"
      >
        <p
          className="author__bio"
        >
          Test bio
          <a
            className="author__bio-twitter"
            href="https://www.linkedin.com/in/undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>
              Test name
            </strong>
             en LinkedIn
          </a>
        </p>
      </div>
    </div>
    <div
      className="post__comments"
    />
  </div>
</div>
`;
